src/lib/journey/callbacks/_utilities/callback.utilities.ts

```diff
@@ -271,32 +329,64 @@ function convertCharacterSetToRuleObj(set: string) {
   const type = arr[1];
 
   if (type === '0123456789') {
-    return {
-      length: Number(num),
-      message: interpolate('minimumNumberOfNumbers', { num: String(num) }),
-      rule: 'numbers',
-    };
+    if (num === '0') {
```

Comment:
All these new checks handle the rule when the num is `0`. Of course, we can't communicate to the user that this requires a character set, but say at least `0` are needed. So, this handles this case with a bit more elegance.